src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/AnyOS/AsnHelpers.cs

```diff
@@ -134,7 +139,18 @@ internal static AlgorithmIdentifier ToPresentationObject(this AlgorithmIdentifie
                     break;
             }
 
-            return new AlgorithmIdentifier(new Oid(asn.Algorithm), keyLength);
+            AlgorithmIdentifier algorithmIdentifier = new AlgorithmIdentifier(new Oid(asn.Algorithm), keyLength);
+            algorithmIdentifier.Parameters = parameters;
+
+            return algorithmIdentifier;
+        }
+
+        private static void GetParameters(AlgorithmIdentifierAsn algorithmIdentifierAsn, ref  byte[] parameters)
```

Comment:
@bartonjs I'm not sure that I go right way with that changes. It would be good if you point me the right code/unit tests to fix.

Reply:
In something like https://github.com/dotnet/corefx/blob/master/src/System.Security.Cryptography.Pkcs/tests/EnvelopedCms/ContentEncryptionAlgorithmTests.cs#L338-L346, I'd do

*  declare `static partial void CheckParameters(AlgorithmIdentifier identifier, int? length)` in the file
* make a ContentEncryptionAlgorithmTests.netcoreapp.cs as a conditional include
* make it implement the partial, something like
```C#
static partial void CheckParameters(AlgorithmIdentifier identifier, int? length)
{
    if (length.HasValue)
    {
        Assert.NotNull(identifier.Parameters);
        Assert.Equal(length.Value, identifier.Parameters.Length);
    }
    else
    {
        Assert.Null(identifier.Parameters);
    }
}
```
* Call CheckParameters(identifier, whatever) from each of the VerifyAlgorithm methods.
* Make a different CheckParameters overload which takes the hex or byte[] or whatever of the expected value
* In the _FixedValue tests in ContentEncryptionAlgorithmTests, call the "I expect this specific value" instead of "I expect this length to be generated" version.